### ISSUE TITLE
signing on ledger nano s hangs during "Select USB Wallet"

### ISSUE THREAD

**Issue by stefanwouldgo** [cryptoadvance/specter-desktop#169]
Trying to sign a pending transaction, I get stuck at the 

```
Select USB Wallet
undefined 
 Detecting... 
```

window. Ledger is unlocked and in Bitcoin app (latest firmware and app). Interestingly, in the terminal, specter keeps saying:

```
[2020-06-22 11:17:12,427] INFO in _internal: 127.0.0.1 - - [22/Jun/2020 11:17:12] "POST /hwi/detect/ HTTP/1.1" 200 -
[{'type': 'ledger', 'model': 'ledger_nano_s', 'path': '0001:0010:00', 'fingerprint': 'xxxxxx', 'needs_pin_sent': False, 'needs_passphrase_sent': False}]
type  not found
```

**Comment by ben-kaufman** [comment]
Hi @stefanwouldgo, which Ledger firmware are you using on the device?

**Comment by stefanwouldgo** [comment]
> Hi @stefanwouldgo, which Ledger firmware are you using on the device?

secure element 1.6, microcontroller 1.12

**Comment by ben-kaufman** [comment]
Thanks, that looks fine. Also, which version of the Bitcoin app? You can check by entering the app and going to the side twice.

**Comment by stefanwouldgo** [comment]
> Thanks, that looks fine. Also, which version of the Bitcoin app? You can check by entering the app and going to the side twice.

1.4.2

**Comment by ben-kaufman** [comment]
Thanks. I believe the issue is due to the SegWit related vulnerability disclosed not long ago, making the latest Trezor firmware and Ledger Bitcoin App (version 1.4.2) incompatible with [HWI](https://github.com/bitcoin-core/HWI).
The issue is currently being handled in [this PR](https://github.com/bitcoin-core/HWI) and we will release a new version once that PR is ready.
We will update here and in the [Telegram group](https://t.me/spectersupport) once the issue is resolved. Meanwhile, if this cannot wait for the HWI patch, let me know and I will help you find an alternative solution.

**Comment by stefanwouldgo** [comment]
Thank you for the instant response. I can wait a little for this to be resolved. BTW, the PR link in your message only leads to the HWI repository.

Thanks for making such a fantastic tool!

**Comment by ben-kaufman** [comment]
Sure! 
Right, the link was wrong, I edited it, this is the PR: https://github.com/bitcoin-core/HWI/pull/340

**Comment by ben-kaufman** [comment]
@stefanwouldgo just a heads up that this issue should now be fixed with the latest version of Specter. Could you please upgrade the version and try again?

**Comment by stefanwouldgo** [comment]
> @stefanwouldgo just a heads up that this issue should now be fixed with the latest version of Specter. Could you please upgrade the version and try again?

Thanks, I had seen the update. Unfortunately I still get a segmentation fault from pip3 when trying to upgrade. This appears to be some problem with python3-pip under ubuntu. I'll let you know when I have managed to update.

**Comment by stefanwouldgo** [comment]
OK, I can confirm the issue is fixed in version 0.5.1. Thank you for the quick help. 

For future reference: upgrading Ubuntu from 18.04 to 19.10/20.04 fixed the pip3 segfault problem during update.
